app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -50,25 +56,33 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        currentAlgorithm = this.getPreferences(Context.MODE_PRIVATE)
+
         imageViewModel.isCameraPermissionGranted = cameraPermissionGranted()
         tryStartCamera()
 
         setContent {
+            val scaffoldState = rememberScaffoldState()
```

Comment:
Variables should be as deep as possible in the Compose structure so they can't be accessed from where they aren't needed. See the key point at the bottom of [this section](https://developer.android.com/jetpack/compose/state#state-hoisting).

And I also thing that you shouldn't put anything outside of your theme as it won't have an effect this way.

Reply:
Fixed. Thanks for the link.